### PR TITLE
Add support for customizing password broker

### PR DESCRIPTION
**Pull request by billmn**
- [X] Changes have been thoroughly tested to not break existing functionality.
- [ ] New functionality has been documented or existing documentation has been updated to reflect changes.
- [ ] Visual changes are explained in the PR description using a screenshot/recording of before and after.

### CODE REVIEW

**Comment by danharrin** [comment]
The default broker should be `null` please, and all types should use `?string`

**Comment by billmn** [comment]
@danharrin Done 👍

File: packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php [hunk 1]
```diff
@@ -63,8 +63,9 @@ public function request(): void
         }
 
         $data = $this->form->getState();
+        $broker = Filament::getAuthPasswordBroker();
```
Comment (zepfietje) [inline packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php]:
```suggestion
        $broker = Filament::getAuthPasswordBroker();
        $data = $this->form->getState();
```

File: packages/panels/src/Pages/Auth/PasswordReset/ResetPassword.php [hunk 2]
```diff
@@ -78,11 +78,12 @@ public function resetPassword(): ?PasswordResetResponse
         }
 
         $data = $this->form->getState();
+        $broker = Filament::getAuthPasswordBroker();
```
Comment (zepfietje) [inline packages/panels/src/Pages/Auth/PasswordReset/ResetPassword.php]:
```suggestion
```

File: packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php [hunk 3]
```diff
@@ -62,9 +62,10 @@ public function request(): void
             return;
         }
 
+        $broker = Filament::getAuthPasswordBroker();
         $data = $this->form->getState();
 
-        $status = Password::sendResetLink(
+        $status = Password::broker($broker)->sendResetLink(
```
Comment (zepfietje) [inline packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php]:
```suggestion
        $status = Password::broker(Filament::getAuthPasswordBroker())->sendResetLink(
```

File: packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php [hunk 4]
```diff
@@ -62,9 +62,10 @@ public function request(): void
             return;
         }
 
+        $broker = Filament::getAuthPasswordBroker();
```
Comment (zepfietje) [inline packages/panels/src/Pages/Auth/PasswordReset/RequestPasswordReset.php]:
```suggestion
```

File: packages/panels/src/Pages/Auth/PasswordReset/ResetPassword.php [hunk 5]
```diff
@@ -82,7 +82,7 @@ public function resetPassword(): ?PasswordResetResponse
         $data['email'] = $this->email;
         $data['token'] = $this->token;
 
-        $status = Password::reset(
+        $status = Password::broker($broker)->reset(
```
Comment (zepfietje) [inline packages/panels/src/Pages/Auth/PasswordReset/ResetPassword.php]:
```suggestion
        $status = Password::broker(Filament::getAuthPasswordBroker())->reset(
```

File: packages/panels/src/Panel/Concerns/HasAuth.php [hunk 6]
```diff
@@ -118,6 +120,13 @@ public function authGuard(string $guard): static
         return $this;
     }
 
+    public function authPasswordBroker(?string $name = null): static
```
Comment (zepfietje) [inline packages/panels/src/Panel/Concerns/HasAuth.php]:
```suggestion
    public function authPasswordBroker(?string $broker = null): static
```

File: packages/panels/src/Panel/Concerns/HasAuth.php [hunk 7]
```diff
@@ -118,6 +120,13 @@ public function authGuard(string $guard): static
...
+    {
+        $this->authPasswordBroker = $name;
```
Comment (zepfietje) [inline packages/panels/src/Panel/Concerns/HasAuth.php]:
```suggestion
        $this->authPasswordBroker = $broker;
```

**Comment by zepfietje** [comment]
Only still docs needed.

**Comment by danharrin** [comment]
Thanks!

**Pull request merged**